substrate/frame/support/src/traits/randomness.rs

```diff
@@ -52,3 +56,91 @@ pub trait Randomness<Output, BlockNumber> {
 		Self::random(&[][..])
 	}
 }
+
+/// An object that is able to produce an arbitrary length slice using some initial set of bytes that
+/// will repeat over and over.
+pub struct RepeatingSlice<'a>(&'a [u8], usize);
+impl<'a> RepeatingSlice<'a> {
+	/// Create a new `RepeatingSlice` given some initial set of input bytes.
+	///
+	/// Empty slices are valid inputs as they are used in `Randomness::random_seed`.
+	pub fn new(s: &'a [u8]) -> Self {
+		Self(s, 0)
+	}
+}
+
+impl<'a> Input for RepeatingSlice<'a> {
+	fn remaining_len(&mut self) -> Result<Option<usize>, Error> {
+		Ok(Some(usize::max_value()))
+	}
+
+	fn read(&mut self, mut into: &mut [u8]) -> Result<(), Error> {
+		let data = &self.0;
+		if data.is_empty() {
+			// Empty slices are valid inputs as they are used in `Randomness::random_seed`. The
+			// solution here is to just return from `Input::read` leaving the destination untouched,
+			// but another option is to fill it with zeroes or some other value (e.g. `0..256`) in
+			// the `RepeatingSlice` constructor, though I expect that it would be more useful in
+			// testing to be able to control the `random_seed` output knowing it's a noop.
+			return Ok(())
+		}
+		let off = &mut self.1;
+		while into.len() != 0 {
+			let len = into.len().min(data.len() - *off);
+			(&mut into[..len]).copy_from_slice(&data[..len]);
+			*off = (*off + len) % data.len();
+			into = &mut into[len..];
+		}
+		Ok(())
+	}
+}
+
+/// This is not a real source of Randomness. Only use this for testing.
+#[cfg(test)]
```

Comment:
This means that this only works in this crate, nowhere else.

However, thinking about this again, we should create some different type for "TestRandomness". As otherwise `()` is not really a good default value.

Reply:
ive made this mistake so many times. fixed.

Reply:
I think the whole point of NOT creating a new type is that then you need to know what that type is. If you are building a pallet, and you want some dummy filler for `Randomness`, you can just use `()`.

Not sure what a user can expect from that beyond it being something dummy like this.

Reply:
I don't have a strong opinion here, I was not the initial author of this idea.

Reply:
Yeah with guarding it on `std` it is better. Fine with me